boot.c

```diff
@@ -166,6 +161,19 @@ init_freemem()
   spa_init(freemem_va_start, freemem_size);
 }
 
+/* initialize user stack */
+void
+init_user_stack()
+{
+  // allocated stack pages right below the runtime
+  alloc_pages(vpn(runtime_va_start - EYRIE_USER_STACK_SIZE),
+      EYRIE_USER_STACK_SIZE >> RISCV_PAGE_BITS,
+      PTE_R | PTE_W | PTE_D | PTE_A | PTE_U);
+
+  // prepare user sp
+  csr_write(sscratch, runtime_va_start);
```

Comment:
Ah, I mean this.
we need to write the return value of the setup_start() to sscratch

please see if that's the case